packages/odyssey-react-mui/src/labs/DateField.tsx

```diff
@@ -47,17 +72,144 @@ const DateField = ({
   hasInitialFocus,
   hint,
   id: idOverride,
+  inputRef,
   isDisabled = false,
   isOptional = false,
   isReadOnly,
   label,
+  minDate,
+  maxDate,
   onBlur,
   onChange,
   onFocus,
+  timezone,
   value,
 }: DateFieldProps) => {
+  const errorMap = useOdysseyDateError();
+  const [displayedErrorMessage, setDisplayedErrorMessage] = useState<
+    string | undefined
+  >(errorMessage);
+
+  const debounceTimeoutRef = useRef<ReturnType<typeof setTimeout>>();
+  const internalValidationError = useRef<string | undefined>(undefined);
+  const localInputRef = useRef<HTMLInputElement>(null);
+
+  useImperativeHandle(
+    inputRef,
+    () => {
+      return {
+        focus: () => {
+          localInputRef.current?.focus();
+        },
+      };
+    },
+    [],
+  );
+
+  useEffect(() => {
+    return () => {
+      if (!debounceTimeoutRef.current) return;
+      clearTimeout(debounceTimeoutRef.current);
+    };
+  }, []);
+
+  const checkMinMaxValidity = useCallback(
+    (value: DateTime) => {
+      const hasMinError = minDate && value.toUTC() < minDate.toUTC();
+      const hasMaxError = maxDate && value.toUTC() > maxDate.toUTC();
+
+      const errorValue = hasMinError
+        ? errorMap.get("minDate")
+        : hasMaxError
+          ? errorMap.get("maxDate")
+          : undefined;
+
+      setDisplayedErrorMessage(errorValue);
+
+      return hasMinError || hasMaxError;
+    },
+    [errorMap, minDate, maxDate],
```

Comment:
I'd prefer this be an `if`-`else-if` because of the way we do other things in general at Okta.

There's a general discouragement against nested ternaries in JS. Why? Readability? Debugging? Formatting? No clue, but it's one of those faux-pas. I don't mind doing an `if`-`else-if`.

It's your choice here.

Here's what I'd like to see instead though:

```suggestion
      setDisplayedErrorMessage(
        hasMinError
          ? errorMap.get("minDate")
          : hasMaxError
            ? errorMap.get("maxDate")
            : undefined
      );

      return hasMinError || hasMaxError;
    },
    [errorMap, minDate, maxDate],
```